src/components/SearchBar/SearchPopup.js

```diff
@@ -47,21 +47,18 @@ function SearchPopup() {
   const id = open ? "simple-popper" : undefined;
 
   const handleSearchChange = (e) => {
-    if (!e.target.value) {
-      history.replace("");
-    } else {
+    if (e.key === "Enter") {
       history.replace(`/search-course/?search=${e.target.value}`);
+      setAnchorEl(null);
+      e.preventDefault();
+      setSearch(e.target.value);
```

Comment:
@saloni-080601 If you are updating the state on onChange then why are you updating here again?

Reply:
I forget to remove it from here

Reply:
Okay, then remove it from here